src/centralserver/admin-ui/src/views/Settings/ApiKeys/ApiKeys.vue

```diff
@@ -0,0 +1,343 @@
+      return this.$store.getters[StoreTypes.getters.HAS_PERMISSION](
+        Permissions.MOCK_PERMISSION1,
+      );
+    },
+    headers(): DataTableHeader[] {
+      return [
+        {
+          text: this.$t('apiKey.table.header.id') as string,
+          align: 'start',
+          value: 'id',
+          class: 'xrd-table-header ts-table-header-server-code',
+        },
+        {
+          text: this.$t('apiKey.table.header.roles') as string,
+          align: 'start',
+          value: 'roles',
+          class: 'xrd-table-header ts-table-header-valid-from',
+        },
+
+        {
+          text: '',
+          value: 'button',
+          sortable: false,
+          class: 'xrd-table-header mr-table-header-buttons',
+        },
+      ];
+    },
+  },
+  created(): void {
+    this.loadKeys();
+  },
+
+  methods: {
+    loadKeys(): void {
+      // Implement loading later
+    },
+    editKey(apiKey: ApiKey): void {
+      this.selectedKey = apiKey;
+      this.selectedRoles = [...this.selectedKey.roles];
+      this.showEditDialog = true;
+    },
+    showRevokeDialog(apiKey: ApiKey): void {
+      this.selectedKey = apiKey;
+      this.confirmRevoke = true;
+    },
+    createApiKey(): void {
+      this.$router.push({
+        name: RouteName.CreateApiKey,
+      });
+    },
+    translateRoles(roles: string[]): string[] {
+      return !roles
+        ? []
+        : roles.map((role) => this.$t(`apiKey.role.${role}`) as string);
+    },
+    async revokeApiKey() {
+      if (!this.selectedKey) return;
+
+      this.removingApiKey = true;
+      return api
+        .remove(`/api-keys/${api.encodePathParameter(this.selectedKey.id)}`)
+        .then((response) => {
+          const key = response.data as ApiKey;
+          this.$store.dispatch(
+            'showSuccessRaw',
+            this.$t('apiKey.table.action.revoke.success', {
+              id: key.id,
+            }),
+          );
+        })
+        .catch((error) =>
+          this.$store.dispatch(StoreTypes.actions.SHOW_ERROR, error),
+        )
+        .finally(() => {
+          this.confirmRevoke = false;
+          this.removingApiKey = false;
+          this.loadKeys();
+        });
+    },
+    async save() {
+      if (!this.selectedKey) return;
+      this.savingChanges = true;
+      return api
+        .put(
+          `/api-keys/${api.encodePathParameter(this.selectedKey.id)}`,
+          this.selectedRoles,
+        )
+        .then((response) => {
+          const key = response.data as ApiKey;
+          this.$store.dispatch(
+            'showSuccessRaw',
+            this.$t('apiKey.table.action.edit.success', {
+              id: key.id,
+            }),
+          );
+        })
+        .catch((error) =>
+          this.$store.dispatch(StoreTypes.actions.SHOW_ERROR, error),
+        )
+        .finally(() => {
+          this.savingChanges = false;
+          this.showEditDialog = false;
+          this.loadKeys();
+        });
+    },
```

Comment:
I don't necessarily see the point of `async` because the promise is being resolved manually in the function. Should it be removed?

Changed the function to use async/await but it actually adds couple more rows to the file in this case 🤷‍♂️
```suggestion
    async save() {
      if (!this.selectedKey) return;
      this.savingChanges = true;
      try {
        const response = await api.put<ApiKey>(
          `/api-keys/${api.encodePathParameter(this.selectedKey.id)}`,
          this.selectedRoles,
        );
        const apiKey = response.data;
        await this.$store.dispatch(
          'showSuccessRaw',
          this.$t('apiKey.table.action.edit.success', {
            id: apiKey.id,
          }),
        );
      } catch (error) {
        await this.$store.dispatch(StoreTypes.actions.SHOW_ERROR, error);
      }

      this.savingChanges = false;
      this.showEditDialog = false;
      this.loadKeys();
    },
```

Reply:
This is some kind of relic from ancient time. Removed it.